### PR TITLE
Migrate package controller test for submit requests

### PR DESCRIPTION
**Pull request by bgeuken**

### CODE REVIEW

**Comment by bgeuken** [comment]
@mdeniz Please review again;-)

File: src/api/spec/controllers/webui/package_controller_spec.rb [hunk 1]
```diff
@@ -38,15 +39,74 @@
         </resultlist>')
   end
 
-  describe 'submit_request' do
+  describe "POST #submit_request" do
+    RSpec.shared_examples "a response of a successfull submit request" do
```
Comment (mdeniz) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: s/sucessfull/sucessful/

File: src/api/spec/controllers/webui/package_controller_spec.rb [hunk 2]
```diff
@@ -38,15 +39,74 @@
...
         post :submit_request, { project: source_project, package: source_package, targetproject: target_project.name }
       end
 
       it { expect(flash[:error]).to eq('Unable to submit: The source of package home:tom/my_package is broken') }
-      it { expect(BsRequestActionSubmit.where(target_project: target_project, target_package: source_package).count).to eq(0) }
+      it { expect(response).to redirect_to(package_show_path(project: source_project, package: source_package)) }
+      it { expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: source_package.name)).not_to exist }
+    end
+
+    context "a submit request that fails due to validation errors" do
```
Comment (mdeniz) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: I think we can do in the same way with shared_examples but passing a param with the string for the `flash[:error]` expectation.

https://www.relishapp.com/rspec/rspec-core/v/2-0/docs/example-groups/shared-example-group#passing-parameters-to-a-shared-example-group

Comment (bgeuken) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: I don't like that. There are to many variations. The tests would become unreadable. Like, we would have to pass the expected flash error, redirect destination, package name, project name and probably more.

File: src/api/spec/controllers/webui/package_controller_spec.rb (hunk 1 above)
Comment (bgeuken) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: Fixed the typo. Thanks!

File: src/api/spec/controllers/webui/package_controller_spec.rb (hunk 2 above)
Comment (mdeniz) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: Oh!, I didn't realize that were different redirections... that's ok then

File: src/api/spec/controllers/webui/package_controller_spec.rb (hunk 1 above)
Comment (marcus-h) [inline src/api/spec/controllers/webui/package_controller_spec.rb]: On 2016-08-19 02:27:08 -0700, Björn Geuken wrote:
> > @@ -38,15 +39,74 @@
> >          </resultlist>')
> >    end
> >  
> > -  describe 'submit_request' do
> > +  describe "POST #submit_request" do
> > +    RSpec.shared_examples "a response of a successfull submit request" do
> 
> Fixed the typo. Thanks!
> 
Nit-pick: the commit message should be "successfull -> successful":)
(commit 96a924a3e41014226d51d2420d0bbaed6c6d1f56)

**Pull request merged**